src/ctapipe/tools/calculate_pixel_stats.py

```diff
@@ -171,13 +173,21 @@ def start(self):
                 f"/dl1/monitoring/telescope/{self.output_table_name}/tel_{tel_id:03d}",
                 overwrite=self.overwrite,
             )
-
-    def finish(self):
         self.log.info(
             "DL1 monitoring data was stored in '%s' under '%s'",
             self.output_path,
             f"/dl1/monitoring/telescope/{self.output_table_name}",
         )
+
+    def finish(self):
+        # Store the subarray description in the output file
+        self.subarray.to_hdf(self.output_path, overwrite=self.overwrite)
+        self.log.info(
+            "Subarray description was stored in '%s'",
+            self.output_path,
+        )
+        # Close the file in the TableLoader
+        self.input_data.close()
```

Comment:
Better use `self.input_data = self.enter_context(TableLoader(...))` when opening.

The tool as an `ExitStack` to manage contexts.